pilot/pkg/bootstrap/sidecarinjector.go

```diff
@@ -47,6 +47,10 @@ const (
 // Was not used in Pilot for 1.3/1.4 (injector was standalone).
 // In 1.5 - used as part of istiod, if the inject template exists.
 func (s *Server) initSidecarInjector(args *PilotArgs) error {
+	if args.ServiceAddress == "" {
```

Comment:
Same question - Should n't this be mandatory?

Reply:
If the variable is not set this would crash. We cannot ensure it is set. We could provide a default I guess, but it would require hardcoding `istiod.istio-system`

Reply:
Ok. Sorry. I thought we are defaulting it some where. Do you see any downside of defaulting it to `istiod.istio-system` so that we not have to check every time we use it?